streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -267,25 +277,32 @@ public long approximateNumEntries() {
 
     @Override
     public void flush() {
+        validateStoreOpen();
         lock.writeLock().lock();
         try {
+            validateStoreOpen();
             cache.flush(cacheName);
-            super.flush();
+            wrapped().flush();
```

Comment:
Isn't this the same thing?

Reply:
They do the same thing, but the majority of calls are on wrapped().